beacon_chain/networking/eth2_network.nim

```
@@ -540,12 +550,16 @@ proc disconnect*(peer: Peer, reason: DisconnectionReason,
           SeenTablePenaltyError
       peer.network.addSeen(peer.peerId, seenTime)
       await peer.network.switch.disconnect(peer.peerId)
-  except CatchableError:
-    # We do not care about exceptions in disconnection procedure.
-    trace "Exception while disconnecting peer", peer = peer.peerId,
-                                                reason = reason
+  except CancelledError as exc:
+    raise exc
+  except CatchableError as exc:
+    # switch.disconnect shouldn't raise
+    warn "Unexpected error while disconnecting peer",
```

Don't we use warning for more actionable information - we are telling the users that there is something wrong with their setup or that we have observed something wrong happening with the network which may explain poor performance, etc. Here, we seem to be telling them about a potential bug in our implementation or alternatively about a very benign networking issue that should not affect their node in any way.

Previously, this was a trace-level message, perhaps `DEBUG` is justified, but warning seems too much.

---

all these warnings would start showing if libp2p starts raising new unexpected errors - it's a transition measure while libp2p upgrades to chronosv4

---

There is no benefit in using `warn` vs `debug` for these kind of messages. They create a risk that we'll run into an unintended increase in log volume (to potentially dangerous levels depending on the particular issue that triggers the increase).

---

#5822 - `warn` would increase chances that we notice it - swallowed exceptions have historically been a source of several security issues - anyway, hopefully we can be rid of them completely soon.